### PR TITLE
Fixed date helper timezone setting bug (#9381)

### PR DESCRIPTION
**Pull request by zce**
#9381

### CODE REVIEW

**Review by aileen — APPROVED** [review]
This fix works fine for me!

File: core/server/helpers/date.js [hunk 1]
```diff
@@ -13,13 +13,14 @@ module.exports = function (date, options) {
...
+    timezone = options.data.blog.timezone;
 
-        // set to published_at by default, if it's available
-        // otherwise, this will print the current date
-        if (this.published_at) {
-            date = moment(this.published_at).tz(timezone).format();
-        }
+    // set to published_at by default, if it's available
+    // otherwise, this will print the current date
+    if (this.published_at) {
```
Comment (kirrg001) [inline core/server/helpers/date.js]: @AileenCGN I haven't tested it , but maybe one case is not supported anymore with this code changes? 

The date helper accepts custom dates and custom formats. Before this change, the condition asked if "no custom date is passed", then we take `published_at` as fallback or now. But now we always prefer `published_at`? Maybe i am wrong, just had a quick look.

Comment (AileenCGN) [reply core/server/helpers/date.js]: 🤔 You might be right here. Testing this case now.

Comment (kirrg001) [reply core/server/helpers/date.js]: 👍

Comment (AileenCGN) [reply core/server/helpers/date.js]: Ok. This PR does indeed only fix one case. Any other date would always be overwritten with the `published_at`. The code, how it was before actually had the correct logic. The only thing that was missing, is the `timezone` information when the helper is called with options.

Comment (kirrg001) [reply core/server/helpers/date.js]: Could you pls PR the correct change? THX!

**Pull request merged**